### ISSUE TITLE
"Clean up" job queues

### ISSUE THREAD

**Issue by sshetenhelm** [yalelibrary/YUL-DC#2884]
**Story**
We should examine and "clean up" our job queues for our various jobs. This may include investigating the following:

- What are the priority weights for all jobs, and do those weights make sense in the current job landscape? 
- Can we limit workers to accepting only up to "X" number of certain job types, or a total of "X" weight for all jobs on a worker?

**Acceptance**

- [x] Identify and document priority weights for all jobs
- [x] Decide if weight changes should be made and, if so, make changes
- [x] Investigate if workers can be limited to only "X" number of certain job types
- [x] Investigate if workers can be limited to a total weight of "X" for all jobs on a worker
- [x] Make necessary changes or, if work is complicated, create new tickets for individual pieces of work

**Comment by jpengst** [comment]
![Image](https://github.com/user-attachments/assets/12a67522-38d2-4bf3-8cd6-da82264a07a9)

**Comment by sshetenhelm** [comment]
FOR YOUR CONSIDERATION, comments and dissenions welcome. 

To be **lowered** in priority: 

- GeneratePDFJob - Reduce from 50 to 10, puts on par with GeneratePtiffJob

To be **raised** in priority: 

The following should be raised to 10 (is that an option?)
- SyncFromPreservicaJob

In theory, this would allow Preservica Syncs to be slightly more prioritized than, say, resyncing an entire collection's metadata.

The following should all be raised to 50 - 
- ExportParentMetadataCsvJob
- ReassociateChildOidsJob
- GenerateOutputCsvJob 
- ExportAllParentSourcesCsvJob 
- ProblemReportManualJob

Basically, with limited exceptions, any job that is user generated or just results in a spreadsheet should be prioritized. 

- UpdatePermissionRequestsJob - **Is this the one that checks to make sure people have or do not have permissions to view on object?** If that is correct, we'd like this bumped up to 100, as **in my opinion, it should be one of the more important jobs.**

**Comment by jpengst** [comment]
Is this the one that checks to make sure people have or do not have permissions to view on object?  
  
Yep, that is correct.  

PR for the adjustments: https://github.com/yalelibrary/yul-dc-management/pull/1433

**Comment by K8Sewell** [comment]
Deployed to Test with release [v2.71.9](https://github.com/yalelibrary/yul-dc-management/releases/tag/v2.71.9)

**Comment by K8Sewell** [comment]
On Test, a sample of jobs are completing successfully - except Preservica Resync jobs which is to be expected.  Not really possible to test the changed priorities effect on how quickly jobs are processed without a significant amount of jobs running.  The ones I tested went really quickly.  Screenshots below of updated priority reflected in GoodJob Dashboard and successfully completed jobs.

Will promote to UAT.

![Image](https://github.com/user-attachments/assets/ba999d6b-99c5-4ce1-8ef8-45d937276f67)

![Image](https://github.com/user-attachments/assets/f23d6aca-a483-4760-9da3-662f892d70c9)

**Comment by sshetenhelm** [comment]
Could someone please remind me which job exports all parent objects by set? I [triggered that one](https://collections-uat.library.yale.edu/management/batch_processes/2217) as well as a 'by source' and the 'by source' worked, but I'm assuming the 'by set' is still in the queue after a bunch of manifest refreshes. 

The 'export all parents by admin set' should also be set at 50.

**Comment by jpengst** [comment]
The 'export all parents by admin set' is handled by the `CreateParentOidCsvJob` and is currently set to -100. Will put back to In Progress for the quick change

**Comment by K8Sewell** [comment]
Deployed to Test with release [v2.72.1](https://github.com/yalelibrary/yul-dc-management/releases/tag/v2.72.1)

**Comment by K8Sewell** [comment]
On Test 'export all parents by admin set' job is set to 50 as expected.  Will promote to UAT.

![Image](https://github.com/user-attachments/assets/281e1b85-3f8c-4f1f-8f4e-3bb60560dabb)

**Comment by sshetenhelm** [comment]
Looks good to me! Ready to close
